pyomo/opt/results/results_.py

```diff
@@ -29,12 +29,18 @@
 logger = logging.getLogger(__name__)
 
 
+def _guess_format(filename):
+    return {'.json': 'json', '.jsn': 'json', '.yaml': 'yaml', '.yml': 'yaml'}.get(
```

Comment:
Can you add a short docstring on why this is necessary? For future / non-John readers.